lib/spdy-transport/protocol/http2/parser.js

```diff
@@ -154,9 +154,18 @@ Parser.prototype.onFrameBody = function onFrameBody(header, buffer, callback) {
   else if (header.type === frameType.X_FORWARDED_FOR)
     this.onXForwardedFrame(header, buffer, callback);
   else
-    callback(null, { type: 'unknown: ' + header.type });
+    this.onUnknownFrame(header, buffer, callback);
 };
 
+Parser.prototype.onUnknownFrame = function onUnknownFrame(header, buffer, callback) {
```

Comment:
There is another thing that fixed by this:
```
HEADERS (without (END_HEADERS/FIN)? flag)
UNKNOWN (fail here)
```
There is no `CONTINUATION`-related frames, I think.
Also, unknown-frames are valid to process anywhere else (just ignore it).

About `CONTINUATION` - `queueHeaderBlock` seems to check that you talk about.